src/tools/Repl/Program.cs

```diff
@@ -0,0 +1,601 @@
+﻿// <copyright file="Program.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.PowerFx;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Types;
+
+namespace PowerFxHostSamples
+{
+    public static class ConsoleRepl
+    {
+        private static RecalcEngine _engine;
+        private static bool _formatTable = true;
+        private const string OptionFormatTable = "FormatTable";
+
+        private static void ResetEngine()
+        {
+            Features toenable = 0;
+            foreach (Features feature in (Features[])Enum.GetValues(typeof(Features)))
+            {
+                toenable |= feature;
+            }
+
+            var config = new PowerFxConfig(toenable);
+
+            config.AddFunction(new HelpFunction());
+            config.AddFunction(new ResetFunction());
+            config.AddFunction(new ExitFunction());
+            config.AddFunction(new OptionFunction());
+            config.AddFunction(new ResetImportFunction());
+            config.AddFunction(new ImportFunction());
+
+            var optionsSet = new OptionSet("Options", DisplayNameUtility.MakeUnique(new Dictionary<string, string>()
+                                                {
+                                                        { OptionFormatTable, OptionFormatTable },
+                                                }));
+
+            config.AddOptionSet(optionsSet);
+
+            _engine = new RecalcEngine(config);
+        }
+
+        public static void Main()
+        {
+            var enabled = new StringBuilder();
+
+            ResetEngine();
+
+            var version = typeof(RecalcEngine).Assembly.GetName().Version.ToString();
+            Console.WriteLine($"Microsoft Power Fx Console Formula REPL, Version {version}");
+
+            foreach (Features feature in (Features[])Enum.GetValues(typeof(Features)))
+            {
+                if ((_engine.Config.Features & feature) == feature && feature != Features.None)
+                {
+                    enabled.Append(" " + feature.ToString());
+                }
+            }
+
+            if (enabled.Length == 0)
+            {
+                enabled.Append(" <none>");
+            }
+
+            Console.WriteLine($"Experimental features enabled:{enabled}");
+
+            Console.WriteLine($"Enter Excel formulas.  Use \"Help()\" for details.");
+
+            REPL(Console.In, false);
+        }
+
+        // Pattern match for Set(x,y) so that we can define the variable
+        public static bool TryMatchSet(string expr, out string arg0name, out FormulaValue varValue)
```

Comment:
This removes a hacky Set regex and uses "Official" APIs. 


Also see
https://github.com/microsoft/Power-Fx/discussions/889